ssl/statem/statem_lib.c

```diff
@@ -1956,6 +1952,8 @@ int ssl_choose_client_version(SSL *s, int version, RAW_EXTENSION *extensions)
  * @s: The SSL connection
  * @min_version: The minimum supported version
  * @max_version: The maximum supported version
+ * @real_version: The maximum version the library is capable of regardless of
+ *                what is enabled for this connection
```

Comment:
This comment could  be more clear. The actual `real_version` is a subtle beast.  It lies just below the lowest **compile-time** hole that lies above at least one **run-time enabled** protocol.  Thus disabling protocols via `SSL_OP_NO_...`at run-time can only **increase** `real_version` by removing all the protocols below some compile-time hole, bringing a new compile-time ceiling into scope.

Fallback adjustments that involves "raising the floor" is not terribly likely, so in practice fallback will have the same `real_version` as what you'd get pre-fallback.